File: submissions/1302-delete-characters-to-make-fancy-string/solution.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        s += ' '
        curr = ''
        count = 0
        ret = ''
        for c in s:
            if curr == c:
                count += 1
            else:
                ret += curr * min(2, count)
                curr = c
                count = 1

        return ret

File: submissions/1302-delete-characters-to-make-fancy-string/solution.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string ret = "";
        int count = 0;
        char curr = '-';
        for (auto c : s) {
            if (c == curr) {
                count += 1;
            } else {
                for (int i = 0; i < min(2, count); i ++) {
                    ret += curr;
                }
                curr = c;
                count = 1;
            }
        }
        for (int i = 0; i < min(2, count); i ++) {
            ret += curr;
        }
        return ret;
    }
};
